build-support/bin/native/bootstrap.sh

```diff
@@ -95,17 +103,21 @@ function bootstrap_native_code() {
   # Bootstraps the native code and overwrites the native_engine_version to the resulting hash
   # version if needed.
   local native_engine_version="$(calculate_current_hash)"
```

Comment:
hm. I'd like there to be tests around how the bootstrap could fail. eg asserting that changes to files triggers recompiling, or failing to compile should result in a missing binary / early exit. Things are still pretty early for this part of the code, so it might make sense to leave them unspecified. But, it's something I'm always thinking about.